data-prepper-plugins/s3-source/src/main/java/org/opensearch/dataprepper/plugins/source/S3ScanPartitionCreationSupplier.java

```diff
@@ -47,7 +47,7 @@ public List<PartitionIdentifier> apply(final Map<String, Object> globalStateMap)
 
         for (final ScanOptions scanOptions : scanOptionsList) {
             final List<String> excludeItems = new ArrayList<>();
-            final S3ScanKeyPathOption s3ScanKeyPathOption = scanOptions.getBucketOption().getkeyPrefix();
+            final S3ScanKeyPathOption s3ScanKeyPathOption = scanOptions.getBucketOption().getS3ScanFilter();
```

Comment:
Overall this looks good, but this will still continuously scan right? Can we add a check for if the scheduling config is added here, and if so to not run this supplier?